controllers/application/deployment.go

```diff
@@ -44,6 +48,13 @@ func (r *ApplicationReconciler) reconcileDeployment(ctx context.Context, applica
 		}
 
 		r.SetLabelsFromApplication(ctx, &deployment, *application)
+		
+		// add an external link to argocd
+		ingresses := application.Spec.Ingresses
+		if len(ingresses) > 0 {
+			deployment.ObjectMeta.Annotations = make(map[string]string, 1)
+			deployment.ObjectMeta.Annotations[AnnotationKeyLinkPrefix] = fmt.Sprintf("https://%s", ingresses[0])
+		}
```

Comment:
This will overwrite any other Annotations a deployment has set. I'm not sure if argo or any other framework in our cluster adds annotations, but it could be preferable to avoid overwriting just in case. We had this issue with labels earlier, so it could show up.